Fix SCDocs assets caching

capital-G opened the pull request:
<!-- For information about contributing see: https://github.com/supercollider/supercollider/wiki/Contributing-directory -->
Merge note: should be squashed

Fixes https://github.com/supercollider/supercollider/issues/6496

## Purpose and Motivation

Currently the caching of static files does not react to a mismatch of `scdoc_version` file but is invalidated by comparing the `mtime` of the static files.
As https://github.com/supercollider/supercollider/pull/6095 changed some static files this will now create problems if a `develop` build is launched (which will replace the static files in the HelpTargetDir as their mtime is newer) and afterwards an existing SuperCollider installation of `3.13` is launched: because the static files are considered "up-to-date" by `3.13` (the mtime of the `HelpSource` file is lower than the one in the `HelpTargetDir`) they will not be copied/replaced, although they are not matching the actual `SCDoc.version` - this create some displaying problems (see #6496).

This PR ties the static files to `SCDoc.version`, so in case we want to update any static files we can do so by incrementing `SCDoc.version` which will invalidate now all files on a version-mismatch (and not just `schelp` files).

Because we can not change the caching-code in older SuperCollider instances (we want SC `<=3.13` to not consider any static files from a newer version valid) it was also necessary to put all static files into a new directory so that any new static files changed via #6095 will not be considered because they are at a different location.
Doing this also helps to clean up the directory though :)

> This will still yield bugs because files like `Browse.html` are still considered "up-to-date" by `<=3.13`, yet the displaying of the help file is not immensly broken (it will be displayed with the newer css theme applied).
> This could only be fixed by changing the locations of `Browse.html` etc. which I consider too drastic, but I am open to input from other users here!

Anyone who had run an build between #6095 and this PR should consider deleting `SCDoc.helpTargetDir` after merging of this PR if switching between `develop` and `stable` is wanted in order to let `<=3.13` copy/use their own static files .

edit: this PR also fixes future modifications of static files

---
I am not too happy with the implementation b/c this introduces side effects to `*checkVersion`, yet I found it the most clean way b/c it is a bit hard to reason about which entrypoints exist to build the docs.
This should catch any changes in `SCDoc.version` at its core.

<!-- Please provide a description, even if you are linking to a related Issue or PR. -->
<!-- If this fixes an open issue, link to it by writing "Fixes #555." -->

## Types of changes

<!-- Delete lines that don't apply -->

- Documentation
- Bug fix

## To-do list

<!-- Complete an item by checking it: [x]. Add new entries to track your progress -->

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review <!-- If not ready for review, consider making a Draft PR instead, or describe what kind of specific feedback you want. -->

<!-- ## Merge notes
(optional) You may also add notes of:
     - Dependencies: if this PR depends on any other PRs or actions
     - Merge Note: any considerations regarding merging this PR
     - Release Notes: text you think would be useful to include in the Release Notes
-->

<!-- Note: The recommended commit message format is:
     component: sub-component: short description of changes
For example:
     classlib: ClassBrowser: fix search with empty query string
Common `comp:` labels:
     help, classlib, sclang, scsynth, plugins, tests, docs, SCDoc
See wiki for more info: Wiki->Creating Pull Requests->git-commit-message-format
-->

prko commented:
Thanks for this, I am satisfied with the theme switcher implemented by you.
Is there a possibility of including the idea or implementation in https://github.com/supercollider/supercollider/pull/6207#issuecomment-2148536975?

capital-G commented:
@tedmoore can you take a look at this? it is not the most fun PR, but it holds me back to using a dev build more often :)

tedmoore commented:
Sorry I haven't had a chance to look closely at this yet (I'm a bit short on time currently). Ahead of the meeting tomorrow I figured I would at least respond to you (unfortunately I can't be a the meeting though)! What I think would be useful is to chat with you sometime so you can walk me through the issue more closely. It's been a couple years since I was intimate with the SCDocs rendering system and I don't think I had trouble with versioning docs. It sounds to me like a more drastic solution might be a better path forward.

capital-G commented:
> What I think would be useful is to chat with you sometime so you can walk me through the issue more closely.

Sure - hmu in the forum chat (dscheiba) and we can figure out a meeting where I can walk you through the PR.

capital-G commented on HelpSource/Overviews/Documents.html:
```diff
@@ -4,18 +4,18 @@
+    <link rel='stylesheet' id='scdoc-theme' href='./../static/themes/default.css' type='text/css' />
+    <link rel='stylesheet' href='./../frontend.css' type='text/css' />
+    <link rel='stylesheet' href='./../custom.css' type='text/css' />
+    <script src="./../staticc/lib/jquery.min.js"></script>
```
probably not call this `staticc`?

tedmoore replied:
Oh I see you've found it too!

capital-G commented:
I'll rebase for the review

tedmoore commented on HelpSource/Overviews/Documents.html:
```diff
@@ -4,18 +4,18 @@
+    <link rel='stylesheet' id='scdoc-theme' href='./../static/themes/default.css' type='text/css' />
+    <link rel='stylesheet' href='./../frontend.css' type='text/css' />
+    <link rel='stylesheet' href='./../custom.css' type='text/css' />
+    <script src="./../staticc/lib/jquery.min.js"></script>
```
This `static` has two `c`s? I believe it's a typo.

tedmoore reviewed (changes requested):
Just the one type in `Documents.html` where there is an extra c in "static"

capital-G commented:
> Just the one type in `Documents.html` where there is an extra c in "static"

Fixed that and rebased.

If everything is fine by you, feel free to squash-merge it :)

prko commented:
Today, I found #6859 and #6860 while testing #6854.
Are they related to this?

On macOS, I have been using a build from last summer and have been manually updating the necessary parts from some commits. However, I could not do it more with some recent commits, and I think this could be related to this PR. Am I correct?
